### PR TITLE
chore: improve the implementation of pubsub module

### PR DESCRIPTION
**Pull request by bzp2010**
### Description

According to the closed #6995 PR, there are some review comments on the pubsub module, fix it throughout the PR.

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -31,8 +31,8 @@ jobs:
           - linux_openresty_1_17
         test_dir:
           - t/plugin
-          - t/admin t/cli t/config-center-yaml t/control t/core t/debug t/discovery t/error_page t/misc t/pubsub
-          - t/node t/router t/script t/stream-node t/utils t/wasm t/xds-library t/xrpc
+          - t/admin t/cli t/config-center-yaml t/control t/core t/debug t/discovery t/error_page t/misc
+          - t/node t/pubsub t/router t/script t/stream-node t/utils t/wasm t/xds-library t/xrpc
```
[the remark this answers is not in the document]
Comment (bzp2010) [reply .github/workflows/build.yml:35]: I'm a little confused, according to the principle you mentioned at https://github.com/apache/apisix/pull/6995#discussion_r869835933, `p` should be in the middle of `n` and `r` in alphabetical order except for `plugin`. lm**n**o**p**q**r**st

File: apisix/core/pubsub.lua [hunk 2]
```diff
@@ -119,8 +138,8 @@ end
 -- no error exists.
 --
 -- @function core.pubsub.on
--- @tparam string command to add callback
--- @tparam function handler callback on receipt of command
+-- @tparam string The command to add callback.
```
[the remark this answers is not in the document]
Comment (bzp2010) [reply apisix/core/pubsub.lua]: fixed

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bzp2010) [reply .github/workflows/build.yml:35]: fixed

File: apisix/core/pubsub.lua [hunk 3]
```diff
@@ -58,23 +58,42 @@ local function init_pb_state()
...
+        end
+    end
+end
+
+
+-- send generic response to client
+local function send_resp(ws, sequence, data)
+    data.sequence = sequence
+    local ok, encoded = pcall(pb.encode, "PubSubResp", data)
+    if not ok or not data then
```
Comment (tzssangglass) [inline apisix/core/pubsub.lua:77]: `not data`?

File: apisix/core/pubsub.lua [hunk 4]
```diff
@@ -119,8 +138,8 @@ end
 -- no error exists.
 --
 -- @function core.pubsub.on
--- @tparam string command to add callback
--- @tparam function handler callback on receipt of command
+-- @tparam string command The command to add callback.
+-- @tparam func handler The callback function on receipt of command.
```
Comment (tzssangglass) [inline apisix/core/pubsub.lua:142]: `-- @tparam function` ?

File: apisix/core/pubsub.lua (hunk 3 above)
Comment (bzp2010) [reply apisix/core/pubsub.lua:77]: A careless mistake that I will fix

File: apisix/core/pubsub.lua (hunk 4 above)
Comment (bzp2010) [reply apisix/core/pubsub.lua:142]: image

The documentation indicates that `func` should be used.

File: apisix/core/pubsub.lua (hunk 3 above)
Comment (bzp2010) [reply apisix/core/pubsub.lua]: @tzssangglass fixed

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**